### DISCUSSION TITLE
[Bug]: Erro na exibição ao atualizar registro do campo tipo Money

### DISCUSSION THREAD

**Discussion by leonardyrj** [leandrocfe/filament-ptbr-form-fields#61] [Q&A]
### What happened?

Estou tendo problema em usar o campo Money quando meu campo é do tipo float.

Se o valor é 10.53 quando edita o formulário ele retorna o valor correto 10,53.
Se o valor é 10.50 quando edita o formulário ele retorna 1,05

Esse comportamento ocorre por causa desse método

```
protected function sanitizeState(?string $state): ?int
    {
        $state = Str::of($state)
            ->replace('.', '')
            ->replace(',', '')
            ->toInteger();

        return $state ?? null;
    }

```
A primeira opção ele retorna 1053 que é o correto para colocar na biblioteca Money

A segunda opção ele retorna 105, dessa forma quando coloca na biblioteca Money retorna errado.
O correto seria retornar 1050

### How to reproduce the bug

Crie um campo do tipo float, e coloque o valor 10,50.
Atualize o registro e será exibido 1,05

### Package Version

3.1.0

### PHP Version

8.2.0

### Laravel Version

11

### Which operating systems does with happen with?

macOS, Linux

**Comment by fabricyo** [comment]
O mesmo erro também está acontecendo comigo, o meu projeto, escrito da mesma forma, porém usando Laravel 10, não tem esse erro. Então estou achando ser algo com o laravel 11.

Edit: 
O meu erro foi corrigido na última versão
https://github.com/leandrocfe/filament-ptbr-form-fields/releases/tag/3.2.1

**Comment by jportiz55** [comment]
Acabei de passar pelo mesmo problema.
Depois de algumas horas consegui resolver alterando todos os campos do money para int e trabalhando com ->intFormat(true). 
Coloquei uma função no helpers pra tratar sempre dividindo por 100 para exibir corretamente.

Resolvi trabalhar com essa lógica após descobrir que bancos utilizam sempre salvam no banco de dados como centavos (int) pra prevenir bugs com double e float.
https://www.pullrequest.com/blog/precision-matters-why-using-cents-instead-of-floating-point-for-transaction-amounts-is-crucial/
